packages/blockchain/ada/src/lib/services/ada.utils.ts

```diff
@@ -0,0 +1,234 @@
+import {
+  Address,
+  BigNum,
+  Bip32PrivateKey,
+  hash_transaction,
+  LinearFee,
+  make_vkey_witness,
+  Transaction,
+  TransactionBody,
+  TransactionBuilder,
+  TransactionBuilderConfigBuilder,
+  TransactionHash,
+  TransactionInput,
+  TransactionOutput,
+  TransactionWitnessSet,
+  Value,
+  Vkeywitnesses,
+} from '@emurgo/cardano-serialization-lib-nodejs'
+import BigNumber from 'bignumber.js'
+import { AdaTransaction, AdaUTXO, BlockchainAdaService } from '@tatumio/api-client'
+
+interface FromAddress {
+  address: string
+  signatureId?: string
+  privateKey?: string
+}
+
+interface FromUTXO {
+  txHash: string
+  index: number
+  signatureId?: string
+  privateKey?: string
+}
+
+const addInput = (transactionBuilder: TransactionBuilder, utxo: AdaUTXO, address: string) => {
+  transactionBuilder.add_input(
+    Address.from_bech32(address),
+    TransactionInput.new(TransactionHash.from_bytes(Buffer.from(utxo.txHash!, 'hex')), utxo.index!),
+    Value.new(BigNum.from_str(utxo.value!)),
+  )
```

Comment:
We should get rid of these ^^^ `non-null assertion`.

Reply:
will do after the type will be fixed, these values should be required

Reply:
Is that OpenApi issue listed somewhere? Also pls add TODO comments

Reply:
no, I was going to write it in the ticket description, I will add comments as well